### ISSUE TITLE
New Rule: No CommonJS in ES modules

### ISSUE THREAD

**Issue by rtsao** [eslint/eslint#17218]
### Rule details

Prevent CommonJS in ES modules

### Related ECMAScript feature

ES modules

### What type of rule is this?

Warns about a potential problem

### Example code

```js
import foo from "foo"; // import statement is used, so this is an ES module

const bar = require("bar"); // INVALID: "ReferenceError: require is not defined in ES module scope, you can use import instead"

module.exports = bar; // INVALID: "ReferenceError: module is not defined in ES module scope"

import { createRequire } from 'node:module';
const constructedRequire = createRequire(import.meta.url);

const baz = constructedRequire("baz"); // OK: constructed require works in an ES module
```

### Why should this rule be in the core instead of a plugin?

- This constraint is rigidly enforced by Node.js, so this rule is potentially applicable to all projects using Node.js
- This rule prevents `ReferenceError` runtime errors:
  - `ReferenceError: module is not defined in ES module scope`
  - `ReferenceError: require is not defined in ES module scope, you can use import instead`

### Participation

- [X] I am willing to submit a pull request to implement this rule.

**Comment by rtsao** [comment]
Accidental duplicate https://github.com/eslint/eslint/issues/17219
